app/models/messaging/bsnl/dlt_template.rb

```diff
@@ -0,0 +1,108 @@
+class Messaging::Bsnl::DltTemplate
+  MAX_VARIABLE_LENGTH = 30
+  TRIMMABLE_VARIABLES = %i[facility_name patient_name].freeze
+  BSNL_TEMPLATES = YAML.load_file("config/data/bsnl_template_details.yml")
+  STATUSES = {
+    "0" => :pending_naming,
+    "1" => :approved
+  }
+
+  attr_reader :name
+  attr_reader :id
+  attr_reader :keys
+  attr_reader :max_length_permitted
+  attr_reader :non_variable_text_length
+  attr_reader :variable_length_permitted
+  attr_reader :status
+
+  def initialize(dlt_template_name)
+    @name = dlt_template_name
+    details = template_details(dlt_template_name)
+    @id = details["Template_Id"]
```

Comment:
We could, keeping them consistent with the BSNL API is lesser indirection though. We'll have to track the mapping between this awkward case and the snakecased version somewhere otherwise.

Reply:
Oh I see. We use those keys to construct the request payload too? In that case, as you were 👍